Add ability to inspect generated SAML response for SAML Clients

thomasdarimont opened the pull request:
Fixes #30328

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

mposolda commented:
@thomasdarimont Would you mind to continue on this PR? I think it can be nice as a feature to Keycloak.

I believe it can be nice to polish a bit (EG. removing comments, make sure that it is shown just for SAML clients and not for OIDC clients etc). And make sure there are not test failures. WDYT?

thomasdarimont committed:
Add SAML response inspection for clients with scope mappings

Add ability to inspect generated SAML response for SAML Clients with client scope mappings.
We also consider default SAML scopes mappers as well as custom scope mappings.

Fixes #30328

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>
(cherry picked from commit b014dd5251a2db4910d87c8db54b3c329006d742)
Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

thomasdarimont commented:
@mposolda I eventually found the time to complete the feature :) 

I think this is now ready for review.

mposolda commented:
@thomasdarimont Sorry for late response :-) Do you please have a chance to rebase and fix conflicts if still interested in this PR?

thomasdarimont commented:
Closing in favour of https://github.com/keycloak/keycloak/pull/49740
